search/search_repository.go

```diff
@@ -205,9 +206,8 @@ func getSearchQueryFromURLString(url string) string {
 }
 
 // parseSearchString accepts a raw string and generates a searchKeyword object
-func parseSearchString(rawSearchString string) searchKeyword {
+func parseSearchString(rawSearchString string) (searchKeyword, error) {
 	// TODO remove special characters and exclaimations if any
-	rawSearchString = strings.ToLower(rawSearchString)
 	rawSearchString = strings.Trim(rawSearchString, "/") // get rid of trailing slashes
 	rawSearchString = strings.Trim(rawSearchString, "\"")
 	parts := strings.Fields(rawSearchString)
```

Comment:
We are doing a url.QueryUnescape() after splitting, so spaces could be passed in as "+".